overlord/hookstate/ctlcmd/set.go

```diff
@@ -165,3 +165,52 @@ func (s *setCommand) setInterfaceSetting(context *hookstate.Context, plugOrSlot
 	attrsTask.Set(which, attributes)
 	return nil
 }
+
+func copyAttributes(value map[string]interface{}) (map[string]interface{}, error) {
+	var cpy interface{}
+	if err := copyRecursive(value, &cpy); err != nil {
+		return nil, err
+	}
+	return cpy.(map[string]interface{}), nil
+}
+
+func copyRecursive(value interface{}, out *interface{}) error {
```

Comment:
I'm not sure why this is not just  copyRecursive(value interface{}) (cpy interface{}, error) ?